go/service/assets/api_op_ListAssets.go

```diff
@@ -15,11 +17,27 @@ type AssetOutput struct {
 	AssetProperties map[string]string `json:"properties"`
 }
 
-func (c *Client) ListAssets() ([]AssetOutput, error) {
-	endpoint := fmt.Sprintf("%s/assets", *c.Addresses.Management)
+func (c *Client) ListAssets(args ...apivalidator.QueryInput) ([]AssetOutput, error) {
+	var queryInput apivalidator.QueryInput
+
+	if len(args) > 0 {
+		queryInput = args[0]
+		err := apivalidator.ValidateQueryInput(args[0].SortOrder)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		queryInput = apivalidator.QueryInput{}
+	}
+
+	endpoint := fmt.Sprintf("%v/assets/request", *c.Addresses.Management)
 	assets := []AssetOutput{}
+	listAssetsQueryJson, err := json.Marshal(queryInput)
 
-	req, err := http.NewRequest(http.MethodGet, endpoint, nil)
+	if err != nil {
+		return nil, fmt.Errorf("unexpected error while marshaling list assets query: %v", err)
```

Comment:
Same as above, I highly recommend updating the error handling method to the new one. You can see an example in line 42.

Reply:
Taken into consideration